overlord/snapstate/snapmgr.go

```diff
@@ -330,14 +330,14 @@ func updateInfo(st *state.State, snapst *SnapState, channel string, userID int,
 	return res[0], nil
 }
 
-func snapInfo(st *state.State, name, channel string, userID int, flags Flags) (*snap.Info, error) {
+func snapInfo(st *state.State, name, channel string, rev snap.Revision, userID int, flags Flags) (*snap.Info, error) {
```

Comment:
yes, we certainly can. The code doesn't do that today, and I don't think I should fix it in this same commit however. But I can stop the madness for the bits I'm changing.
```
snapd$ egrep -r '\b(rev|revno)\b' * | wc -l
131
snapd$ egrep -lr '\b(rev|revno)\b' * | wc -l
25
```